src/components/shop/shop.js

```diff
@@ -52,9 +65,27 @@ export class Shop extends Screen {
         this.events.once("shutdown", scaleEvent.unsubscribe);
     }
 
+    setVisible(menu) {
+        if (menu === "top") {
+            this.menus.top.setVisible(true);
+            this.menus.shop.setVisible(false);
+            this.menus.inventory.setVisible(false);
+            eventBus.removeSubscription(this.customMessage);
+            eventBus.subscribe(this.backMessage);
+            return;
+        }
+
+        eventBus.removeSubscription(this.backMessage);
+        this.menus.top.setVisible(false);
+        menu === "shop" && this.menus.shop.setVisible(true);
+        menu === "manage" && this.menus.inventory.setVisible(true);
+        eventBus.subscribe(this.customMessage);
+    }
+
```

Comment:
this feels a bit like two functions in one depending on the string

Reply:
Definitely needed a refactor. Down from 16 lines to 6! 9a8904e